### PR TITLE
DOC: Document howto build documentation in a virtual environment

### LINKED ISSUE #18811: Cannot build documentation without successful "import numpy" error when having built numpy to a virtual enviornment
## Documentation

Issue in file `doc/source/docs/howto_build_docs.rst`
Upon building numpy in a virtual environment and then trying to build documentation, `make html` and other document building commands throw an error saying:
```
numpy not found, cannot build documentation without successful "import numpy"
make: *** [Makefile:90: version-check] Error 1
```
This happens because `Makefile` defines `PYTHON=python$(PYVER)` which when executed later, runs  `make html` command using the global python installation instead of using the virtual environment (yes, even when the current shell session has the virtual environment activated).  It could be documented in the `howto_build_docs.rst` to override `Makefile`'s `PYTHON` at the command line while running `make html` by simply changing it to `make PYTHON=python html`.

### PR DESCRIPTION
**Pull request by verma16Ayush**
Closes #18811

### CODE REVIEW

File: doc/source/docs/howto_build_docs.rst [hunk 1]
```diff
@@ -48,13 +48,22 @@ with::
...
+    cd doc
     make html
 
 in the ``doc/`` directory. If all goes well, this will generate a
 ``build/html`` subdirectory containing the built documentation. If you get
 a message about ``installed numpy != current repo git version``, you must
 either override the check by setting ``GITVER`` or re-install NumPy.
 
+Note that if you have built numpy into a virtual enviornment and getting
+an error that says ``numpy not found, cannot build documentation without successful "import numpy"``
```
Comment (charris) [inline doc/source/docs/howto_build_docs.rst:60]: Please keep the lines < 80 characters.

Comment (charris) [reply doc/source/docs/howto_build_docs.rst:60]: Maybe use ``numpy not found, cannot build documentation without . . .``

Comment (verma16Ayush) [reply doc/source/docs/howto_build_docs.rst:60]: Thanks for the assist @charris, I'll make these changes, although could you please clarify what you mean by

> Maybe use ` ``numpy not found, cannot build documentation without . . .`` `

should I replace ` ``numpy not found, cannot build documentation without successful "import numpy"`` ` with exactly what you said?

**Comment by melissawm** [comment]
Hello, @verma16Ayush - thanks for this! In the PR description, if you include the words "Closes #18811" or "Fixes #18811", the issue will automatically be closed when the PR is merged. [Here´s a complete list of words that do that](https://docs.github.com/pt/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

**Comment by verma16Ayush** [comment]
Thanks @melissawm I really didn't know that 😅. 
Still fresh to oss. I'll add it as soon as i get a go from @charris with the changes.

File: doc/source/docs/howto_build_docs.rst (hunk 1 above)
Comment (charris) [reply doc/source/docs/howto_build_docs.rst:60]: That is the suggestion, the ellipsis indicates that text has been omitted. I looked for ways to break inline literals and could not find any and even putting it on a new line was a tiny bit over, although probably acceptable. The other option was to change the error message itself, but that seemed overkill.

File: doc/source/docs/howto_build_docs.rst [hunk 2]
```diff
@@ -48,13 +48,19 @@ with::
...
 
+    cd doc
     make html
 
 in the ``doc/`` directory. If all goes well, this will generate a
 ``build/html`` subdirectory containing the built documentation. If you get
 a message about ``installed numpy != current repo git version``, you must
 either override the check by setting ``GITVER`` or re-install NumPy.
 
+Note that if you have built numpy into a virtual enviornment and getting an error that says ``numpy not found, cannot build documentation without...`` you need to override the makefile ``PYTHON`` variable at the commandline, so instead of writing ``make html`` write::
```
Comment (charris) [inline doc/source/docs/howto_build_docs.rst:59]: The lines need to be < 80 characters in length. You need hard line breaks.

Comment (verma16Ayush) [reply doc/source/docs/howto_build_docs.rst]: I only just realised what I was doing wrong, I guess? Is it okay now? Thanks for walking me through though.

**Comment by charris** [comment]
Thanks @verma16Ayush .

**Pull request merged**